rocksdb/src/main/java/com/yahoo/ycsb/db/rocksdb/RocksDBClient.java

```diff
@@ -0,0 +1,387 @@
+          deserializeValues(iterator.value(), fields, values);
+          result.add(values);
+          iterations++;
+        }
+      }
+
+      return Status.OK;
+    } catch(final RocksDBException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status update(final String table, final String key, final Map<String, ByteIterator> values) {
+    //TODO(AR) consider if this would be faster with merge operator
+
+    try {
+      if (!COLUMN_FAMILIES.containsKey(table)) {
+        createColumnFamily(table);
+      }
+
+      final ColumnFamilyHandle cf = COLUMN_FAMILIES.get(table).getHandle();
+      final Map<String, ByteIterator> result = new HashMap<>();
+      final byte[] currentValues = rocksDb.get(cf, key.getBytes(UTF_8));
+      if(currentValues == null) {
+        return Status.NOT_FOUND;
+      }
+      deserializeValues(currentValues, null, result);
+
+      //update
+      result.putAll(values);
+
+      //store
+      rocksDb.put(cf, key.getBytes(UTF_8), serializeValues(result));
+
+      return Status.OK;
+
+    } catch(final RocksDBException | IOException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status insert(final String table, final String key, final Map<String, ByteIterator> values) {
+    try {
+      if (!COLUMN_FAMILIES.containsKey(table)) {
+        createColumnFamily(table);
+      }
+
+      final ColumnFamilyHandle cf = COLUMN_FAMILIES.get(table).getHandle();
+      rocksDb.put(cf, key.getBytes(UTF_8), serializeValues(values));
+
+      return Status.OK;
+    } catch(final RocksDBException | IOException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+  }
+
+  @Override
+  public Status delete(final String table, final String key) {
+    try {
+      if (!COLUMN_FAMILIES.containsKey(table)) {
+        createColumnFamily(table);
+      }
+
+      final ColumnFamilyHandle cf = COLUMN_FAMILIES.get(table).getHandle();
+      rocksDb.delete(cf, key.getBytes(UTF_8));
+
+      return Status.OK;
+    } catch(final RocksDBException e) {
+      e.printStackTrace();
+      return Status.ERROR;
+    }
+  }
+
+  private void saveColumnFamilyNames() throws IOException {
+    final Path file = rocksdbDir.resolve(CF_NAMES_FILENAME);
+    try(final PrintWriter writer = new PrintWriter(new OutputStreamWriter(Files.newOutputStream(file), UTF_8))) {
+      writer.println(new String(RocksDB.DEFAULT_COLUMN_FAMILY, UTF_8));
+      for(final String cfName : COLUMN_FAMILIES.keySet()) {
+        writer.println(cfName);
+      }
+    }
+  }
+
+  private List<String> loadColumnFamilyNames() throws IOException {
+    final List<String> cfNames = new ArrayList<>();
+    final Path file = rocksdbDir.resolve(CF_NAMES_FILENAME);
+    if(Files.exists(file)) {
+      try (final LineNumberReader reader =
+               new LineNumberReader(new InputStreamReader(Files.newInputStream(file), UTF_8))) {
```

Comment:
instead of `new InputStreamReader(Files.newInputStream(file), UTF_8)` you could use `Files.newBufferedReader(file, UTF_8)`

Reply:
Fixed.